### PR TITLE
Add LIP "Add 256-bit Integer Types to Lisk Codec"

### PR DESCRIPTION
**Pull request by AndreasKendziorra**

### CODE REVIEW

File: proposals/lip_add_256_bit_integer_types_to_lisk_codec.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+### Type Conversion to Protobuf
+
+When [converting a Lisk JSON schema to a .proto file](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0027.md#appendix-b-json-schema-to-protobuf), properties of data type _uint256_ or _int256_ in the Lisk JSON schema are converted to fields of type _bytes_ in the `.proto` file.
+
+## Rationale
+
+### Why Byte Representation Instead of Varint?
+
+Using varint encoding would break the compatibility with protobuf heavily. Recall that Lisk codec was designed in a way that [every Lisk JSON schema can be converted to a .proto file](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0027.md#appendix-b-json-schema-to-protobuf) and that every protobuf decoder decodes binary messages in the same way as Lisk codec using this `.proto` file. Using varint encoding would result in encoded values up to 37 bytes. There is, however, no field type in protobuf to which a protobuf decoder could decode a varint encoded value of more than 10 bytes. Hence, one could not convert a Lisk JSON schema that uses _uint256_ or _int256_ to a `.proto` file that could be used by a protobuf decoder for the same binary messages.
```
Comment (vjaiman) [inline proposals/lip_add_256_bit_integer_types_to_lisk_codec.md:80]: i) I would explain why it is `37 bytes`.
ii) Also, I would add a reference of 10 bytes. For example, https://protobuf.dev/programming-guides/encoding/#varints or similar one.

**Review by vjaiman — APPROVED** [review]
Looks all good. I have left a minor comment,

File: proposals/lip_add_256_bit_integer_types_to_lisk_codec.md (hunk 1 above)
Comment (AndreasKendziorra) [reply proposals/lip_add_256_bit_integer_types_to_lisk_codec.md]: I addressed your points.

**Review by gkoumout — APPROVED** [review]
Looks good to me, just left an optional suggestion. 

PS: I did not check the validity of binary messages in appendix, I guess this has been already done.

File: proposals/lip_add_256_bit_integer_types_to_lisk_codec.md [hunk 2]
```diff
@@ -77,7 +77,7 @@ When [converting a Lisk JSON schema to a .proto file](https://github.com/LiskHQ/
...
+Using varint encoding would break the compatibility with protobuf heavily. Recall that Lisk codec was designed in a way that [every Lisk JSON schema can be converted to a .proto file](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0027.md#appendix-b-json-schema-to-protobuf) and that every protobuf decoder decodes binary messages in the same way as Lisk codec using this `.proto` file. Using varint encoding would result in encoded values up to 37 bytes, because varint encoding puts 7 bits of the number into a byte. Hence, ⌈256/7⌉ bytes are required to encode a 256-bit integer. There is, however, no field type in protobuf to which a protobuf decoder could decode a varint encoded value of more than 10 bytes. This is because the [supported number types](https://protobuf.dev/programming-guides/proto2/#scalar) do not use more than 64 bits. Hence, one could not convert a Lisk JSON schema that uses _uint256_ or _int256_ to a `.proto` file that could be used by a protobuf decoder for the same binary messages.
```
[the remark this answers is not in the document]
Comment (AndreasKendziorra) [reply proposals/lip_add_256_bit_integer_types_to_lisk_codec.md]: done

**Review by ricott1 — APPROVED** [review]

**Pull request merged**